endpoints/api/organization.py

```diff
@@ -819,3 +820,156 @@ def post(self, orgname, client_id):
+                return "Created", 201
+        except model.DataModelException as e:
+            logger.error("Error while creating Proxy cache configuration as: %s", str(e))
+
+        return request_error("Error while creating Proxy cache configuration")
+
+    @nickname("deleteProxyCacheConfig")
+    def delete(self, orgname):
+        """
+        Delete proxy cache configuration for the organization.
+        """
+        permission = AdministerOrganizationPermission(orgname)
+        if not permission.can():
+            raise Unauthorized()
+
+        try:
+            model.proxy_cache.get_proxy_cache_config_for_org(orgname)
+        except model.InvalidProxyCacheConfigException:
+            raise NotFound()
+
+        try:
+            success = model.proxy_cache.delete_proxy_cache_config(orgname)
+            if success:
+                return "Deleted", 201
+        except model.DataModelException as e:
+            logger.error("Error while deleting Proxy cache configuration as: %s", str(e))
+        raise request_error(message="Proxy Cache Configuration failed to delete")
+
+
+@resource("/v1/organization/<orgname>/validateproxycache")
+@show_if(features.PROXY_CACHE)
+class ProxyCacheConfigValidation(ApiResource):
+    """
+    Resource for validating Proxy Cache Config.
+    """
+
+    schemas = {
+        "NewProxyCacheConfig": {
+            "type": "object",
+            "description": "Proxy cache configuration for an organization",
+            "required": ["upstream_registry"],
+            "properties": {
+                "upstream_registry": {
+                    "type": "string",
+                    "description": "Name of the upstream registry that is to be cached",
+                },
+            },
+        },
+    }
+
+    @nickname("validateProxyCacheConfig")
+    @validate_json_request("NewProxyCacheConfig")
+    def post(self, orgname):
+        permission = AdministerOrganizationPermission(orgname)
+        if not permission.can():
+            raise Unauthorized()
+
+        try:
+            model.proxy_cache.get_proxy_cache_config_for_org(orgname)
+            request_error("Proxy Cache Configuration already exists")
+        except model.InvalidProxyCacheConfigException:
+            pass
+
+        data = request.get_json()
+
+        # filter None values
+        data = {k: v for k, v in data.items() if v is not None}
+
+        try:
+            config = ProxyCacheConfig(**data)
+            existing = model.organization.get_organization(orgname)
+            config.organization = existing
+
+            proxy = Proxy(config, "something-totally-fake", True)
+            response = proxy.get(f"{proxy.base_url}/v2/")
+            if response.status_code == 200:
+                return "Valid", 202
+        except UpstreamRegistryError as e:
+            raise request_error(
+                message="Failed login to remote registry. Please verify entered details and try again."
```

Comment:
An `UpstreamRegistryError` won't always happen in connection with a failed login, so this might be misleading at times.
I think want both to be specific about the error as well as show an user-friendly message I think we need add the upstream response status code to the `UpstreamRegistryError` class so that we can conditionally show a login error or a more generic one (connection related).
We probably want to log `e` too - if more info is needed it should be available in the logs.
All of this should be done a follow-up PR though, this is great as is for now o/